coda-prototype/src/io/StorageUtils.js

```diff
@@ -0,0 +1,196 @@
+class StorageManager {
+    constructor() {
+        /*
+        var manager = this;
+        chrome.storage.local.get("lastEdit", (editObj) => {
+            console.log(editObj);
+            editObj["lastEdit"] = new Date(JSON.parse(editObj["lastEdit"]));
+            console.log(editObj["lastEdit"]);
+            if ( Object.prototype.toString.call(editObj["lastEdit"]) === "[object Date]" ) {
+                if (!isNaN((editObj["lastEdit"]).getTime())) {
+                    // date is in valid format
+                    if (this.isExpired()) {
+                        manager.lastEdit = new Date();
+                        this.clearStorage().then( () => console.log(manager.lastEdit));
+
+                    } else {
+                        manager.lastEdit = editObj["lastEdit"];
+                        console.log(editObj["lastEdit"]);
+                        console.log(manager.lastEdit);
+                    }
+
+                }
+            }
+        });*/
+    }
+    static get instance() {
+        return this._instance || (this._instance = new StorageManager());
+    }
+    isExpired() {
+        // TODO
+        // on every startup of CODA, check if storage is expired, i.e. more than 30 days have passed since last edit
+        // if yes, clear storage
+        var _MS_PER_DAY = 1000 * 60 * 60 * 24;
+        // a and b are javascript Date objects
+        function dateDiffInDays(a, b) {
+            // Discard the time and time-zone information.
+            let utc1 = Date.UTC(a.getFullYear(), a.getMonth(), a.getDate());
+            let utc2 = Date.UTC(b.getFullYear(), b.getMonth(), b.getDate());
+            return Math.floor((utc2 - utc1) / _MS_PER_DAY);
+        }
+        return (30 <= dateDiffInDays(this.lastEdit, new Date()));
```

Comment:
This logic feels backwards compared to how I would read it "the last edit was more than 30 days ago" - can we reverse the conditional?